modules/service/src/main/resources/db/migration/V0927__grouping_modes.sql

```diff
@@ -0,0 +1,216 @@
+      c_grating,
+      c_filter,
+      c_fpu,
+      c_central_wavelength,
+      c_xbin,
+      c_xbin_default,
+      c_ybin,
+      c_ybin_default,
+      c_amp_read_mode,
+      c_amp_gain,
+      c_roi,
+      c_wavelength_dithers,
+      c_spatial_offsets
+    )
+  ) STORED;
+
+ALTER TABLE t_gmos_south_long_slit
+  ADD COLUMN c_mode_key text NOT NULL GENERATED ALWAYS AS (
+    format_gmos_long_slit_mode_group(
+     'gs',
+      c_observing_mode_type,
+      c_grating,
+      c_filter,
+      c_fpu,
+      c_central_wavelength,
+      c_xbin,
+      c_xbin_default,
+      c_ybin,
+      c_ybin_default,
+      c_amp_read_mode,
+      c_amp_gain,
+      c_roi,
+      c_wavelength_dithers,
+      c_spatial_offsets
+    )
+  ) STORED;
+
+-- The observing mode group view contains an entry per distinct observing mode
+-- and program combination across all observing modes.
+CREATE VIEW v_observing_mode_group AS
+  SELECT
+    m.c_mode_key,
+    o.c_program_id,
+    max(m.c_observation_id) as c_observation_id -- arbitrary, just pick one
+  FROM
+    t_gmos_north_long_slit m
+  LEFT JOIN t_observation o ON m.c_observation_id = o.c_observation_id
+  GROUP BY
+    m.c_mode_key,
+    o.c_program_id
+UNION ALL
+  SELECT
+    m.c_mode_key,
+    o.c_program_id,
+    max(m.c_observation_id) as c_observation_id -- arbitrary, just pick one
+  FROM
+    t_gmos_south_long_slit m
+  LEFT JOIN t_observation o ON m.c_observation_id = o.c_observation_id
+  GROUP BY
+    m.c_mode_key,
+    o.c_program_id;
+
+-- Replace the observation view to add an observing mode key.
+DROP VIEW v_observation;
+
+CREATE OR REPLACE VIEW v_observation AS
+  SELECT o.*,
+  CASE WHEN o.c_explicit_ra              IS NOT NULL THEN o.c_observation_id END AS c_explicit_base_id,
+  CASE WHEN o.c_air_mass_min             IS NOT NULL THEN o.c_observation_id END AS c_air_mass_id,
+  CASE WHEN o.c_hour_angle_min           IS NOT NULL THEN o.c_observation_id END AS c_hour_angle_id,
+  CASE WHEN o.c_observing_mode_type      IS NOT NULL THEN o.c_observation_id END AS c_observing_mode_id,
+  CASE WHEN o.c_spec_wavelength          IS NOT NULL THEN o.c_observation_id END AS c_spec_wavelength_id,
+  CASE WHEN o.c_spec_signal_to_noise_at  IS NOT NULL THEN o.c_observation_id END AS c_spec_signal_to_noise_at_id,
+  CASE WHEN o.c_spec_wavelength_coverage IS NOT NULL THEN o.c_observation_id END AS c_spec_wavelength_coverage_id,
+  CASE WHEN o.c_spec_focal_plane_angle   IS NOT NULL THEN o.c_observation_id END AS c_spec_focal_plane_angle_id,
+  CASE WHEN o.c_observation_duration     IS NOT NULL THEN o.c_observation_id END AS c_observation_duration_id,
+  c.c_active_start::timestamp + (c.c_active_end::timestamp - c.c_active_start::timestamp) * 0.5 AS c_reference_time,
+  COALESCE(
+    CASE WHEN o.c_observing_mode_type = 'gmos_north_long_slit' THEN mode_gnls.c_mode_key END,
+    CASE WHEN o.c_observing_mode_type = 'gmos_south_long_slit' THEN mode_gsls.c_mode_key END,
+    NULL
+  ) AS c_mode_key
```

Comment:
This `COALESCE` chunk would be the new bit on the view. It's probably dumb to add an explicit `NULL`.  Apart from that, the issue is that `c_mode_key` isn't on `t_observation` itself but rather in a view.  I might need to add it to `t_observation` so it can be indexed and then keep it up-to-date with triggers?  Advice?